README.md

```diff
@@ -610,3 +630,86 @@ To avoid per-prefix rate limiting with Amazon S3, you may want to try using
 `--s3.key_format=2`, which stores blobs across a larger number of prefixes.
 Reference:
 [Optimizing Amazon S3 Performance](https://docs.aws.amazon.com/AmazonS3/latest/dev/optimizing-performance.html).
+
+## Prometheus Categories
+
+bazel-remote supports declaring categories for the prometheus metrics.
+The categories can be used to calculate cache hit ratio separate per
+type of build, see where most traffic comes from, etc.
+
+Clients can set categories via HTTP and gRPC headers. With bazel
+that is done via the bazel option --remote_header.
+
+Allowed category names are declared in the bazel-remote configuration file.
+The allowed value set for each category also have to be declared as an
+attempt to avoid polluting Prometheus with too many different time series.
+https://prometheus.io/docs/practices/naming/ warns:
+
+> "CAUTION: Remember that every unique combination of key-value
+> label pairs represent a new time series, which can dramatically
+> increase the amount of data stored. Do not use labels to store
+> dimensions with high cardinality (many different label values),
+> such as user IDs, email addresses, or other unbounded sets of
+> values."
+
+Received headers that match a declared category name, but with a value outside
+the declared allowed value set, is reported with the value "other" to
+Prometheus. This is convenient for categories such as "branch", where it from a
+cache hit ratio perspective often make sense to distinguish between "main"
+branch and "other" branches.
```

Comment:
I think we should mention here that only lowercase category names are allowed, otherwise it will likely be a common error that people will hit when first trying this feature. Also: are non lowercase values allowed?